client/src/file/locked_item.rs

```diff
@@ -0,0 +1,16 @@
+use crate::{
+    Key,
+    file::{self, Item, api},
+};
+
+#[derive(Clone, Debug)]
+pub struct LockedItem {
+    pub(crate) inner: api::EncryptedItem,
+}
+
+impl LockedItem {
+    // TODO should return the original item if we fail to decrypt it.
+    pub fn unlock(self, key: &Key) -> Result<Item, file::Error> {
+        self.inner.decrypt(key)
+    }
```

Comment:
I guess that is because we consume self. Why would we want to keep the original locked item if we can't even decrypt it? how can we return both the file::Error and the original item?

Reply:
Yes, we don't want to lose the original item, but considering that there might be no way to unlock it after that point that might be OK?

Reply:
One thing is that if there is an item in a object path, it should be stable, this might be a reason to keep it around even if one cannot do anything with it.

Reply:
Does LockedItem implements Clone? i guess it is fine :)

Reply:
No longer relevant for now